base_local_planner/src/speed_limiters/path_speed_limiter.cpp

```diff
@@ -114,12 +118,13 @@ bool PathSpeedLimiter::calculateLimits(double& max_allowed_linear_vel, double& m
       minimum_distance = dist_from_path;
       // Reset distance
       distance_from_robot = std::max(0.0, segment_length - distanceAlongLineSegment(pos2, p0, p1));
+      path_came_within_min_dist = true;
     }
-    else
+    else if(path_came_within_min_dist)
```

Comment:
would it be incorrect to just check if `distance_from_robot` is > 0 or != 0?

Reply:
I don't think we can since distance_from_robot could be 0 for two consecutive positions when robot is turning.